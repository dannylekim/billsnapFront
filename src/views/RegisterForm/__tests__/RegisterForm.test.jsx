import React from 'react';
import RegisterFormContainer, { RegisterForm, defaultError } from '../RegisterForm.jsx';
import { shallow } from "enzyme";

describe('RegisterForm', () => {
    let wrapper;
    let handleMockFunction;
    let mockToggleFormType;
    let mockOnChange;

    const conditions = [
      {
        name:"confirmPassword",
        condition: false,
        toolTipInfo: {
          open: false,
          errorMessage: ""
        }
      },
      {
        name:"password",
        condition: false,
        toolTipInfo: {
          open: false,
          errorMessage: ""
        }
      },
      {
        name:"firstName",
        condition: false,
        toolTipInfo: {
          open: false,
          errorMessage: ""
        }
      },
      {
        name:"lastName",
        condition: false,
        toolTipInfo: {
          open: false,
          errorMessage: ""
        }
      },
      {
        name:"email",
        condition: false,
        toolTipInfo: {
          open: false,
          errorMessage: ""
        }
      }
    ];

    beforeEach(() => {
      handleMockFunction = jest.fn();
      mockToggleFormType =  jest.fn();
      mockOnChange = jest.fn();
      wrapper = shallow(
        <RegisterForm onChange = {mockOnChange} setFormType = {mockToggleFormType} handleButtonClick={handleMockFunction} validInvalidByName={handleMockFunction} conditions={conditions} alertNotification={defaultError}/>
      );
    });
  
    afterEach(() => {
      handleMockFunction.mockRestore();
    });
  
    describe("render", () => {
      describe("snapshots ðŸ“¸", () => {
        it("Register should match snap shot", () => {
          const firstNameToolTip = document.createElement('div');
          const lastNameToolTip = document.createElement('div');
          const emailToolTip = document.createElement('div');
          const passwordToolTip = document.createElement('div');
          const verifyPasswordToolTip = document.createElement('div');
          
          firstNameToolTip.setAttribute("id", "firstName");
          document.body.appendChild(firstNameToolTip);

          lastNameToolTip.setAttribute("id", "lastName");
          document.body.appendChild(lastNameToolTip);

          emailToolTip.setAttribute("id", "email");
          document.body.appendChild(emailToolTip);

          passwordToolTip.setAttribute("id", "password");
          document.body.appendChild(passwordToolTip);

          verifyPasswordToolTip.setAttribute("id", "confirmPassword");
          document.body.appendChild(verifyPasswordToolTip);

          matches(<RegisterForm onChange = {mockOnChange} setFormType = {mockToggleFormType} handleButtonClick={handleMockFunction} validInvalidByName={handleMockFunction} conditions={conditions} alertNotification={defaultError}/>);
        });

        it("RegisterFormContainer should match snap shot", () => {
          matches(<RegisterFormContainer />)
        })
      });
  
      describe("components", () => {});
    });
  
    describe("functions", () => {

      it("onChange is called when form input is changed", () => {
        wrapper.find("FormGroup").at(0).props().onChange();
        expect(mockOnChange).toBeCalledTimes(1);
      });
  

      it("click formToggle", () => {
        wrapper.find(".form__toggle").simulate("click");
        expect(mockToggleFormType).toBeCalledTimes(1);
      });

    });
});