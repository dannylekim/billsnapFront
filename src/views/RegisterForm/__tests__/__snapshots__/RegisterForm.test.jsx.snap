// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm render snapshots ðŸ“¸ Register should match snap shot 1`] = `
<div
  className="register__container"
>
  <div
    className="hidden__div"
  />
  <img
    alt="character logo"
    className="character__icon__image"
    src="./billSnapIcon.png"
  />
  <form
    className=""
  >
    <div
      className="form__inputs"
    >
      <div
        className="form-group"
        onChange={[MockFunction]}
      >
        <input
          autoComplete="first-name"
          className="register__login__inputs form-control"
          id="firstName"
          name="firstName"
          placeholder="First name"
          type="text"
        />
      </div>
      <div
        className="form-group"
        onChange={[MockFunction]}
      >
        <input
          autoComplete="last-name"
          className="register__login__inputs form-control"
          id="lastName"
          name="lastName"
          placeholder="Last name"
          type="text"
        />
      </div>
      <div
        className="form-group"
        onChange={[MockFunction]}
      >
        <input
          autoComplete="username"
          className="register__login__inputs form-control"
          id="email"
          name="email"
          placeholder="Email"
          type="email"
        />
      </div>
      <div
        className="form-group"
        onChange={[MockFunction]}
      >
        <input
          autoComplete="new-password"
          className="register__login__inputs form-control"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="form-group"
        onChange={[MockFunction]}
      >
        <input
          autoComplete="new-password"
          className="register__login__inputs form-control"
          id="confirmPassword"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <button
        className="login_register__submit__button btn btn-primary btn-md btn-pill"
        name="submit"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </form>
  <div>
    <div
      className="form__seperator"
    >
      <hr
        className="form__horizontal__line"
      />
      Or
      <hr
        className="form__horizontal__line"
      />
    </div>
    <div>
      <h6>
        Have an account?
      </h6>
      <button
        className="form__toggle btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        ðŸ‘‹ Log in to your account.
      </button>
    </div>
  </div>
</div>
`;

exports[`RegisterForm render snapshots ðŸ“¸ RegisterFormContainer should match snap shot 1`] = `
<div
  className="register__container"
>
  <div
    className="hidden__div"
  />
  <img
    alt="character logo"
    className="character__icon__image"
    src="./billSnapIcon.png"
  />
  <form
    className=""
  >
    <div
      className="form__inputs"
    >
      <div
        className="form-group"
        onChange={[Function]}
      >
        <input
          autoComplete="first-name"
          className="register__login__inputs form-control"
          id="firstName"
          name="firstName"
          placeholder="First name"
          type="text"
        />
      </div>
      <div
        className="form-group"
        onChange={[Function]}
      >
        <input
          autoComplete="last-name"
          className="register__login__inputs form-control"
          id="lastName"
          name="lastName"
          placeholder="Last name"
          type="text"
        />
      </div>
      <div
        className="form-group"
        onChange={[Function]}
      >
        <input
          autoComplete="username"
          className="register__login__inputs form-control"
          id="email"
          name="email"
          placeholder="Email"
          type="email"
        />
      </div>
      <div
        className="form-group"
        onChange={[Function]}
      >
        <input
          autoComplete="new-password"
          className="register__login__inputs form-control"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="form-group"
        onChange={[Function]}
      >
        <input
          autoComplete="new-password"
          className="register__login__inputs form-control"
          id="confirmPassword"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <button
        className="login_register__submit__button btn btn-primary btn-md btn-pill"
        name="submit"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </form>
  <div>
    <div
      className="form__seperator"
    >
      <hr
        className="form__horizontal__line"
      />
      Or
      <hr
        className="form__horizontal__line"
      />
    </div>
    <div>
      <h6>
        Have an account?
      </h6>
      <button
        className="form__toggle btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        ðŸ‘‹ Log in to your account.
      </button>
    </div>
  </div>
</div>
`;
